### PR TITLE
plugins/memblaze: Add smart-log, latency and performance features

### PR DESCRIPTION
**Pull request by workingman3008**
Add smart-log, latency and performance features

### CODE REVIEW

File: plugins/memblaze/memblaze-nvme.c [hunk 1]
```diff
@@ -1197,3 +1197,954 @@ static int mb_set_lat_stats(int argc, char **argv,
...
+struct smart_log_add_item_10 {
+    uint8_t id;
+    uint8_t norm;
+    union {
+        struct wear_level wear_level;  // 0xad
+        uint8_t           raw[6];
+    };
+    uint8_t rsvd[2];
+};
+#pragma pack(pop)
```
Comment (igaw) [inline plugins/memblaze/memblaze-nvme.c:1270]: I suggest to use `__packed` attribute here. Those open coded pragmas are usually easy to miss.

**Comment by igaw** [comment]
Check your coding style settings. We follow the kernel style, e.g. tabs instead of spaces at the beginning of the line. The `checkpatch.sh` script from the Linux tree helps to identify offending lines:

```shell
$ ~/work/linux/scripts/checkpatch.pl -g HEAD   
[...]

ERROR: code indent should use tabs where possible
#994: FILE: plugins/memblaze/memblaze-nvme.h:28:
+        ENTRY("lat-stats-print-x", "Get Latency Statistics log and show it.", mb_get_latency_stats)$

WARNING: please, no spaces at the start of a line
#994: FILE: plugins/memblaze/memblaze-nvme.h:28:
+        ENTRY("lat-stats-print-x", "Get Latency Statistics log and show it.", mb_get_latency_stats)$

WARNING: line length of 102 exceeds 100 columns
#995: FILE: plugins/memblaze/memblaze-nvme.h:29:
+		ENTRY("lat-log-print-x", "Get High Latency log and show it.", mb_get_high_latency_log)

WARNING: line length of 102 exceeds 100 columns
#996: FILE: plugins/memblaze/memblaze-nvme.h:30:
+        ENTRY("perf-stats-print-x", "Get Performance Stat log and show it.", mb_get_performance_stats)

ERROR: code indent should use tabs where possible
#996: FILE: plugins/memblaze/memblaze-nvme.h:30:
+        ENTRY("perf-stats-print-x", "Get Performance Stat log and show it.", mb_get_performance_stats)$

WARNING: please, no spaces at the start of a line
#996: FILE: plugins/memblaze/memblaze-nvme.h:30:
+        ENTRY("perf-stats-print-x", "Get Performance Stat log and show it.", mb_get_performance_stats)$

total: 560 errors, 772 warnings, 976 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

NOTE: Whitespace errors detected.
      You may wish to use scripts/cleanpatch or scripts/cleanfile

Commit 219e69226f6c ("plugins/memblaze: Add smart-log, latency and performance features") has style problems, please review.

NOTE: If any of the errors are false positives, please report
```

File: plugins/memblaze/memblaze-nvme.c (hunk 1 above)
Comment (workingman3008) [reply plugins/memblaze/memblaze-nvme.c:1270]: Got, I will create a new thread for this PR.

**Comment by workingman3008** [comment]
I will close this PR and create a new thread on your comments for this PR.

**Pull request closed without merging**